protobuf/ProtoGrainGenerator/Template.cs

```diff
@@ -117,29 +132,25 @@ public async Task ReceiveAsync(IContext context)
                 }    
                 case GrainRequestMessage(var methodIndex, var r):
                 {
+                    Action<string> onError = error => context.Respond( new GrainErrorResponse {Err = error } );
+
                     switch (methodIndex)
                     {
 			            {{#each Methods}}
                         case {{Index}}:
-                        {                            
-                            try
-                            {
-                                var res = await _inner.{{Name}}(({{InputName}})r);
-                                var response = new GrainResponseMessage(res);                                
-                                context.Respond(response);
-                            }
-                            catch (Exception x)
-                            {
-                                var grainErrorResponse = new GrainErrorResponse
-                                {
-                                    Err = x.ToString()
-                                };
-                                context.Respond(grainErrorResponse);
+                        {   
+                            if(r is {{InputName}} input){
+                                await _inner.{{Name}}(input, res => context.Respond( new GrainResponseMessage(res) ), onError);
```

Comment:
`res => context.Respond( new GrainResponseMessage(res) )`

Could be a private instance method if we want to get rid of the closure capture we talked about. 

 await _inner.{{Name}}(input, respond, onError);

....

private void respond<T>(T response)
{
    _context.respond(response);
}
```

Reply:
Not a bad idea, fixing